source/s3_meta_request.c

```diff
@@ -212,6 +211,21 @@ int aws_s3_meta_request_init_base(
     return AWS_OP_SUCCESS;
 }
 
+void aws_s3_meta_request_cancel(struct aws_s3_meta_request *meta_request) {
+    AWS_PRECONDITION(meta_request);
+    AWS_PRECONDITION(meta_request->vtable->cancel);
+    bool cancel_called = true;
+    aws_s3_meta_request_lock_synced_data(meta_request);
```

Comment:
hmmm, need something better. The lock here is a little dangerous. But nothing pop in my head.